corehq/apps/ota/tests/test_utils.py

```diff
@@ -425,3 +429,100 @@ def test_get_restore_user_as_user_for_commcare_user(self):
+        self._create_synclog(self.domain, 'abc123', 'device-id', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 1):
+            self.assertFalse(can_login_on_device('abc123', 'new-device-id'))
+
+    def test_allowed_if_device_count_is_under_limit(self):
+        self._create_synclog(self.domain, 'abc123', 'device-id', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 2):
+            self.assertTrue(can_login_on_device('abc123', 'new-device-id'))
+
+    def test_web_apps_logins_are_always_allowed(self):
+        for i in range(3):
+            self._create_synclog(self.domain, 'abc123', f'WebAppsLogin*{i}', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 1):
+            self.assertTrue(can_login_on_device('abc123', 'WebAppsLogin*newlogin'))
+
+    def test_web_apps_logins_do_not_count_towards_device_count(self):
+        for i in range(3):
+            self._create_synclog(self.domain, 'abc123', f'WebAppsLogin*{i}', date=self.within_past_day)
+
+        self._create_synclog(self.domain, 'abc123', 'device-id', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 2):
+            self.assertTrue(can_login_on_device('abc123', 'new-device-id'))
+
+    def test_activity_older_than_a_day_is_ignored(self):
+        for i in range(2):
+            self._create_synclog(self.domain, 'abc123', f'device-{i}', date=self.over_one_day_ago)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 1):
+            self.assertTrue(can_login_on_device('abc123', 'new-device-id'))
+
+    def test_either_date_or_last_submitted_counts_towards_device_count(self):
+        self._create_synclog(
+            self.domain,
+            'abc123',
+            'device-1',
+            date=self.over_one_day_ago,
+            last_submitted=self.within_past_day,
+        )
+
+        self._create_synclog(
+            self.domain,
+            'abc123',
+            'device-2',
+            date=self.within_past_day,
+            last_submitted=self.over_one_day_ago,
+        )
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 2):
+            self.assertFalse(can_login_on_device('abc123', 'new-device-id'))
+
+    def test_allowed_if_empty_queryset(self):
+        self.assertTrue(can_login_on_device('abc123', 'device-0'))
+
+    def test_allowed_for_different_user(self):
+        self._create_synclog(self.domain, 'abc123', 'device-id', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 1):
+            self.assertTrue(can_login_on_device('def456', 'device-id'))
+
+    def test_device_id_is_none_is_allowed(self):
+        self._create_synclog(self.domain, 'abc123', 'device-id', date=self.within_past_day)
+
+        with patch('corehq.apps.ota.utils.DEVICES_PER_USER', 1):
+            self.assertTrue(can_login_on_device('abc123', 'device-id'))
```

Comment:
Is this meant to be None?
```suggestion
            self.assertTrue(can_login_on_device('abc123', None))
```

Reply:
Good catch: 0b04d75dbb98b3a825ca02e26d47730a47d1ed03